src/app/shared-services/informed-consent/informed-consent.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core'
+import { Observable, of } from 'rxjs'
+import { InformedConsent, FirstTimeConsent } from './default'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class InformedConsentService {
+  constructor() {}
+
+  getInformedConsent(key: string): Observable<InformedConsent> {
+    const value = localStorage.getItem(key)
+    if (value) {
+      return of(JSON.parse(localStorage.getItem(key)))
+    } else {
+      const value2 = new InformedConsent()
+      if (value2) {
```

Comment:
Can `if (value2)` ever fail, since you've just declared it above? If not, than the `if`-clause is obsolete.